src/app/pages/system/enclosure/components/enclosure-side/enclosure-svg/enclosure-svg.component.ts

```diff
@@ -113,16 +122,65 @@ export class EnclosureSvgComponent implements OnDestroy {
   });
 
   private highlightSelectedSlot = effect(() => {
-    Object.values(this.overlayRects).forEach((overlay) => overlay.classList.remove('selected'));
+    const selectedSlot = this.selectedSlot();
+    const slots = this.slots();
+
+    this.clearSelectionStylesFromAllSlots();
 
-    if (!this.selectedSlot()) {
+    if (!selectedSlot) {
       return;
     }
 
-    const slotOverlay = this.overlayRects[this.selectedSlot().drive_bay_number];
-    this.renderer.addClass(slotOverlay, 'selected');
+    this.addSelectedStyles(selectedSlot);
+    this.addSelectedVdevDisksStyles(selectedSlot, slots);
   });
 
+  private clearSelectionStylesFromAllSlots(): void {
+    Object.values(this.overlayRects).forEach((overlay) => {
+      overlay.classList.remove('selected');
+      overlay.classList.remove('selected-vdev-disk');
+      overlay.classList.remove('not-selected-vdev-disk');
+    });
+  }
+
+  private addSelectedStyles(selectedSlot: DashboardEnclosureSlot): void {
+    const slotOverlay = this.overlayRects[selectedSlot.drive_bay_number];
+    this.renderer.addClass(slotOverlay, 'selected');
+  }
+
+  private addSelectedVdevDisksStyles(
+    selectedSlot: DashboardEnclosureSlot,
+    allSlots: DashboardEnclosureSlot[],
+  ): void {
+    const selectedVdevDisks = selectedSlot.pool_info?.vdev_disks.map(
+      (diskInfo) => diskInfo.dev,
+    ).filter(
+      (disk) => disk !== selectedSlot.dev,
+    );
+
+    if (!selectedVdevDisks?.length) {
+      return;
+    }
+    const selectedVdevDisksDriveBayNumbers = [];
+    const notSelectedVdevDisksDriveBayNumbers = [];
+    for (const slot of allSlots) {
+      if (selectedVdevDisks.includes(slot.dev)) {
+        selectedVdevDisksDriveBayNumbers.push(slot.drive_bay_number);
+      } else if (slot.drive_bay_number !== selectedSlot.drive_bay_number) {
+        notSelectedVdevDisksDriveBayNumbers.push(slot.drive_bay_number);
+      }
+    }
+
+    for (const vdevDiskDriveBayNumber of selectedVdevDisksDriveBayNumbers) {
+      const vdevDiskSlotOverlay = this.overlayRects[vdevDiskDriveBayNumber];
+      this.renderer.addClass(vdevDiskSlotOverlay, 'selected-vdev-disk');
+    }
+    for (const notVdevDiskDriveBayNumber of notSelectedVdevDisksDriveBayNumbers) {
+      const vdevDiskSlotOverlay = this.overlayRects[notVdevDiskDriveBayNumber];
+      this.renderer.addClass(vdevDiskSlotOverlay, 'not-selected-vdev-disk');
```

Comment:
Feels like you can simplify this for example by not just updating overlay classes in the first for loop, or by iterating over `overlayRects`.